pgjdbc/src/test/java/org/postgresql/test/jdbc2/ArrayTest.java

```diff
@@ -416,8 +416,7 @@ public void testDirectFieldString() throws SQLException {
     assertEquals(" \fnew \n ", array[2]);
     assertEquals(" ", array[3]);
 
-    // this is likely undesired behavior of dropping white spaces mid-value (not
-    // leading/trailing)
+    // PostgreSQL drops leading and trailing whitespace, so does the driver
```

Comment:
This test asserts that the driver is also dropping ascii white space in the middle of an unquoted value.
This is not consistent with the backend. However, the backend should never produce this type of value.

Reply:
@vlsi thoughts?
This is asserting dropping of mid value white spaces, which is probably undesired, but practically a non issue as the backend should never produce such a value.

Reply:
> This is not consistent with the backend

Why do you think so?

Reply:
Please add a test when the same values get passed to the backend.

Reply:
@vlsi I have added test (https://github.com/pgjdbc/pgjdbc/pull/1266/commits/aae0d487847524c3d37582c99c3ed838608e04ef)

Reply:
It is still different from testDirectFieldString